### PR TITLE
runtime: Enable UDP GRO by default

### PR DESCRIPTION
**Pull request by abeyad**
The runtime guard `envoy.reloadable_features.prefer_quic_client_udp_gro` is now enabled by default.

The Envoy Mobile APIs to set UDP GRO have also been removed, after confirming behavior and performance with experimental results.

### CODE REVIEW

File: mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java
Comment (alyssawilk) [inline mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java:55]: Would it make more sense to change the default value of the flag in runtime_guards.cc and remove the config knobs?
/wait-any

**Comment by abeyad** [comment]
@danzh2010 do you think we can default enable the UDP GRO runtime guard? https://github.com/envoyproxy/envoy/blob/0e6450aac2d0010555c82eb1ff513cf234ab5817/source/common/runtime/runtime_features.cc#L137.  Are there still concerns for server Envoy (i.e. not Envoy Mobile) now that https://github.com/envoyproxy/envoy/issues/33474 is fixed?

File: mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java
Comment (danzh2010) [reply mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java:55]: We need to make the RUNTIME_GUARD default true for sure, but I think this change is also needed until we deprecate the RUNTIME_GUARD.

Comment (danzh2010) [reply mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java:55]: After all, shall we wait for the experiment result before submit this change?

Comment (abeyad) [reply mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java:55]: which experiment are you referring to?

Comment (danzh2010) [reply mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java:55]: We are doing an Android experiment with preferring GRO this week, and I assume we can know the result early next week.

Comment (abeyad) [reply mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java:55]: sure, we can wait till then.  Since I have the PR up, I'll prepare the changes anyway.

RE: server side Envoy, are we comfortable with flipping the runtime guard to true by default?

File: mobile/test/java/io/envoyproxy/envoymobile/engine/EnvoyConfigurationTest.kt
Comment (alyssawilk) [inline mobile/test/java/io/envoyproxy/envoymobile/engine/EnvoyConfigurationTest.kt:269]: do we want to leave this in, given we still want to make sure we explicitly set for now?

**Review by alyssawilk — COMMENTED** [review]
LG with optional nit, once we sort out the plan for default value.

File: mobile/test/java/io/envoyproxy/envoymobile/engine/EnvoyConfigurationTest.kt
Comment (abeyad) [reply mobile/test/java/io/envoyproxy/envoymobile/engine/EnvoyConfigurationTest.kt:269]: good point, re-added

File: mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java
Comment (danzh2010) [reply mobile/library/java/org/chromium/net/impl/NativeCronvoyEngineBuilderImpl.java]: > RE: server side Envoy, are we comfortable with flipping the runtime guard to true by default?

Yes, that's the expected procedure for any default-true envoy flags.

File: source/common/runtime/runtime_features.cc
Comment (danzh2010) [inline source/common/runtime/runtime_features.cc:109]: No need for the TODO, just follow the normal flag deprecation schedule.

**Comment by danzh2010** [comment]
Please update the PR title and description now that it's not mobile only.

File: mobile/test/java/io/envoyproxy/envoymobile/engine/EnvoyConfigurationTest.kt
Comment (abeyad) [reply mobile/test/java/io/envoyproxy/envoymobile/engine/EnvoyConfigurationTest.kt:269]: actually, since we made the flag default true now, this is no longer an applicable check (and causes the test to fail)

**Comment by abeyad** [comment]
> Please update the PR title and description now that it's not mobile only.

done

File: mobile/library/cc/engine_builder.cc
Comment (alyssawilk) [inline mobile/library/cc/engine_builder.cc:860]: We'd discussed leaving this in in case import had concerns?  If you want to remove it today let's either get pre-approval from EPT that they don't have concerns, or you'd have to monitor import to make sure if they want to do delay rollout we add a platform-switch instead of a disable on import.

Comment (danzh2010) [reply mobile/library/cc/engine_builder.cc:860]: I won't insist on keeping the existing API because we have an mobile API to flip a reloadable flag now. We can explicit set the flag there if needed.

**Review by alyssawilk — APPROVED** [review]

**Comment by alyssawilk** [comment]
alas now has conflicts
/wait on merge

**Review by alyssawilk — APPROVED** [review]
LGTM ci-willing

**Pull request merged**
